### PR TITLE
Add otc provider

### PR DESCRIPTION
**Pull request by cgroschupp**
Add Provider for the german cloud provider OTC (https://cloud.telekom.de/)

### CODE REVIEW

**Comment by cgroschupp** [comment]
@marcelcorso I adapted the provider to the new structure, can you merge the provider?

**Comment by marcelcorso** [comment]
@cgroschupp <3 thanks!
@samwierema said he wanted to review

File: provider/otc/otc.go [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	return bytes.NewReader(body1), nil
+}
+
+//Send send sms to n number of people using bulk sms api
+func (c *OTC) Send(message sachet.Message) (err error) {
+	err = c.loginRequest()
```
Comment (marcelcorso) [inline provider/otc/otc.go]: Wouldn't it be better if you did the login on `NewOTC `? 
And then for every `Send` call you would only actually send the sms?  
I think it makes it easier to understand and does less requests. 

The question is how long does a token is valid?

File: provider/otc/otc.go [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/json")
+	if len(d.token) > 0 {
+		req.Header.Set("X-Auth-Token", d.token)
+	}
+
+	tr := http.DefaultTransport.(*http.Transport)
+	tr.TLSClientConfig = &tls.Config{InsecureSkipVerify: d.Insecure}
```
Comment (marcelcorso) [inline provider/otc/otc.go]: The tls cert on https://smn.eu-de.otc.t-systems.com/ is valid. Why are you skipping the verification? 
Not sure this is a good idea in general.

File: provider/otc/otc.go [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+	if err != nil {
+		return err
+	}
+
+	c.otcBaseURL = fmt.Sprintf("https://smn.eu-de.otc.t-systems.com/v2/%s", c.ProjectID)
+
+	return nil
+}
+
+func (d *OTC) SendRequest(method, resource string, payload interface{}) (io.Reader, error) {
```
Comment (marcelcorso) [inline provider/otc/otc.go]: "interface{} says nothing." :P 
you can avoid it by moving the json marshaling out of the function for example.

File: provider/otc/otc.go [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+				Type      string `json:"type"`
+				Endpoints []struct {
+					URL string `json:"url"`
+				} `json:"endpoints"`
+			} `json:"catalog"`
+		} `json:"token"`
+	}
+	var endpointResp endpointResponse
+
+	err = json.NewDecoder(resp.Body).Decode(&endpointResp)
```
Comment (marcelcorso) [inline provider/otc/otc.go]: why do you parse the response into `endpointResponse` if you are not going to use it?

File: provider/otc/otc.go (hunk 3 above)
Comment (marcelcorso) [reply provider/otc/otc.go]: or maybe kill the `SendRequest` function altogether since it's only used once.

Comment (marcelcorso) [reply provider/otc/otc.go]: or maybe kill the `SendRequest` function altogether since it's only used once.

File: provider/otc/otc.go (hunk 2 above)
Comment (cgroschupp) [reply provider/otc/otc.go]: But I only activate InsecureSkipVerify if Insecure is true, is not it ok?

**Comment by cgroschupp** [comment]
@marcelcorso Thanks for the review, i tried to fix everything.

File: provider/otc/otc.go (hunk 2 above)
Comment (marcelcorso) [reply provider/otc/otc.go]: oh, true!

**Comment by marcelcorso** [comment]
looks good. @samwierema ?

**Comment by cgroschupp** [comment]
@samwierema Can you please merge this PR?

**Pull request merged**